### PR TITLE
Feature/probinso/misc fixes

### PR DESCRIPTION
**Pull request by robinson96**

### CODE REVIEW

File: src/chai/ArrayManager.cpp [hunk 1]
```diff
@@ -215,50 +250,54 @@ void ArrayManager::allocate(
...
-          auto alloc = m_resource_manager.getAllocator(
-              pointer_record->m_allocators[space]);
-          alloc.deallocate(space_ptr);
-
-          for (int space_t = CPU; space_t < NUM_EXECUTION_SPACES; ++space_t) {
-            if (space_ptr == pointer_record->m_pointers[space_t])
-              pointer_record->m_pointers[space_t] = nullptr;
-          }
-        } else {
+           if (space_ptr == pointer_record->m_pointers[UM]) {
```
Comment (davidbeckingsale) [inline src/chai/ArrayManager.cpp]: I hope these are not tabs!!!!

Comment (robinson96) [reply src/chai/ArrayManager.cpp]: No, just my editor is set to 3 spaces per tab instead of 2.

Comment (davidbeckingsale) [reply src/chai/ArrayManager.cpp]: 3?!

Comment (rhornung67) [reply src/chai/ArrayManager.cpp]: What!!! 3 spaces?!?

File: src/chai/ArrayManager.cpp [hunk 2]
```diff
@@ -101,16 +101,48 @@ void ArrayManager::registerPointer(
     record->m_owned[i] = true;
   }
   record->m_owned[space] = owned;
+
+  // register with umpire if it's not there, update size info if it is
+  if (pointer) {
+     auto allocation_record = m_resource_manager.findAllocationRecord(pointer);
+     if (allocation_record) {
+        allocation_record->m_size = record->m_size;
```
Comment (davidbeckingsale) [inline src/chai/ArrayManager.cpp]: Is there another way to do this?

File: src/chai/ManagedArray.inl [hunk 3]
```diff
@@ -469,44 +553,96 @@ ManagedArray<T>::operator= (std::nullptr_t from) {
...
-    return;
-  }
-
-  ExecutionSpace const prev_space = m_resource_manager->getExecutionSpace();
-  m_resource_manager->setExecutionSpace(space);
-
-  T_non_const* non_const_active_base_pointer = const_cast<T_non_const*>(m_active_base_pointer);
-  for (int i = 0; i < size(); ++i) {
-    non_const_active_base_pointer[i] = T(m_active_base_pointer[i]);
+ManagedArray<T>::moveInnerImpl()
```
Comment (adayton1) [inline src/chai/ManagedArray.inl]: See what's in develop for the move method. This may be no longer necessary.

**Comment by davidbeckingsale** [comment]
Hey @robinson96 - I think these changes have all gone in under separate PRs from you and @adayton1 . Please let me know if I can close this.

**Comment by robinson96** [comment]
> 
> 
> Hey @robinson96 - I think these changes have all gone in under separate PRs from you and @adayton1 . Please let me know if I can close this.

Yeah, you can close it.

**Pull request closed without merging**